frontend/cypress/integration/components/menu.spec.js

```diff
@@ -0,0 +1,45 @@
+describe('Menu', () => {
+  beforeEach(() => {
+    cy.visit('/');
+  });
+
+  it('navigates to all internal website pages', () => {
+    // Open the menu
+    cy.get('[data-cy=menu-toggle]').click();
+    // Navigate to About page
+    cy.get('[data-cy=about-link]').click();
+    // Check current url
+    cy.url().should('match', /about$/);
+
+    // Open the menu
+    cy.get('[data-cy=menu-toggle]').click();
+    // Navigate to Resources page
+    cy.get('[data-cy=resources-link]').click();
+    // Check current url
+    cy.url().should('match', /resources$/);
+
+    // Open the menu
+    cy.get('[data-cy=menu-toggle]').click();
+    // Navigate to Sponsors page
+    cy.get('[data-cy=sponsors-link]').click();
+    // Check current url
+    cy.url().should('match', /sponsors$/);
+
+    // Open the menu
+    cy.get('[data-cy=menu-toggle]').click();
+    // Navigate to Engage page
+    cy.get('[data-cy=engage-link]').click();
+    // Check current url
+    cy.url().should('match', /engage$/);
+  });
```

Comment:
This seems like something we can put in a for loop and use the .each cypress tag with an array of links for easy refactoring if needed later on.